MoltenVK/MoltenVK/GPUObjects/MVKDeviceMemory.mm

```diff
@@ -38,7 +38,7 @@
 		return mvkNotifyErrorWithText(VK_ERROR_MEMORY_MAP_FAILED, "Memory is already mapped. Call vkUnmapMemory() first.");
 	}
 
-	if ( !ensureHostMemory() ) {
+	if ( !ensureMTLBuffer() && !ensureHostMemory() ) {
```

Comment:
An elegant solution!

I was concerned that large images can require more memory than a `MTLBuffer` can provide...that was a main reason why `MVKDeviceMemory` had to introduce host memory separate from `MTLBuffer` memory...but I see that `ensureMTLBuffer()` already guards against that in this situation...so this should cover both use cases.